libs/pilight/core/common.c

```diff
@@ -111,7 +112,7 @@ void array_free(char ***array, int len) {
 	}
 }
 
-unsigned int explode(char *str, const char *delimiter, char ***output) {
+unsigned int explode(const char *str, const char *delimiter, char ***output) {
```

Comment:
Just to make that clear: changing a function from
```int f(char *p) { ... }```
to
```int f(const char *p) { ... }```
does not add an resctriction of how f() can be called. It adds an enhacement:
```c
int f1(char *p) { ... }
int f2(const char *p) { ... }

char a[] = "...";
const char b[] = "...";

f1(a);
f1(b);  // compiler warning! f1 is resticted to be called with non-const char buffer.
f2(a);
f2(b);  // no striction, no compiler warning.
```
On the other hand, changing a function from
```int *g() { ... }```
to
```const int *g() { ... }```
adds an restriction to how the return value can be stored:
```c
int *g1() { ... }
const int *g2() { ... }

int *a;
const int *b;

a = g1();
b = g1();
a = g2();  // compiler warning: return value is restricted to be stored in a const int* variable.
b = g2();
```